File: 🟡 Medium/240 - Search a 2D Matrix II/_240SearchA2DMatrixII.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m = len(matrix)
        n = len(matrix[0])
        row, column = m - 1, 0

        while row >= 0 and column < n:
            if matrix[row][column] == target:
                return True
            elif matrix[row][column] > target:
                row -= 1
            else:
                column += 1

        return False

File: 🟡 Medium/240 - Search a 2D Matrix II/_240SearchA2DMatrixII.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m = matrix.size();
        int n = matrix[0].size();
        int row = m - 1, column = 0;

        while (row >= 0 && column < n) {
            if (matrix[row][column] == target) {
                return true;
            } else if (matrix[row][column] > target) {
                // move to top
                row--;
            } else {
                // move to right
                column++;
            }
        }

        return false;
    }
};